services/ingestion/etl/agri/csv/src/events/onParseCSV.js

```diff
@@ -1,71 +1,60 @@
-import path from 'path';
-import stream from 'stream';
 import AWS from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
 import parse from 'csv-parse';
 import transform from 'stream-transform';
 
-import transformRecord from '../lib/transform';
-
-// Destination bucket
-const { BUCKET } = process.env;
-
-export const handler = (event, context, callback) => {
-  /*
-   * Some checks here before going any further
-   */
-
-  // Only work on the first record
-  const snsRecord = event.Records ? event.Records[0] : undefined;
+import Logger from '../../../../../../logger/listener/src/lib/Logger';
 
-  // Was the lambda triggered correctly? Is the file extension supported? etc.
-  if (!snsRecord || snsRecord.EventSource !== 'aws:sns') {
-    return callback('Bad record');
-  }
-
-  /*
-   * Prepare file analysis
-   */
+// Import constants
+import { STATUS } from '../../../../../../storage/meta-index/src/events/onStatusReported';
 
-  // Extract message
-  const message = JSON.parse(snsRecord.Sns.Message);
+// Import logic
+import { extractMessage, prepareMessage } from '../lib/sns';
+import transformRecord from '../lib/transform';
+import uploadFromStream from '../lib/uploadFromStream';
 
-  // Check file extension
-  if (path.extname(message.object.key) !== '.csv') {
-    return callback('File extension should be .csv');
-  }
+export const handler = async (event, context, callback) => {
+  // 1. Validate handler execution
+  // check event, context
+  const snsMessage = extractMessage(event);
```

Comment:
Excellent extraction of the `extractMessage` as a small step! ;)
Maybe later it could be abstracted a bit more when it makes sense, but for now it's already taking a lot of bloat and repetition! Thanks